backend/internal/api/service_handlers.go

```diff
@@ -222,14 +223,56 @@ func (h *ServiceHandler) getViewportNodes(c *gin.Context) {
 
 // parseVault triggers a vault parsing operation
 func (h *ServiceHandler) parseVault(c *gin.Context) {
-	// TODO: Implement when VaultService is available
-	c.JSON(http.StatusNotImplemented, gin.H{"error": "Vault parsing not yet implemented"})
+	// Use request context
+	ctx := c.Request.Context()
+
+	// Check if vault service is available
+	if h.vaultService == nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Vault service not configured"})
+		return
+	}
+
```

Comment:
TODO(CL): Add more granular error handling for different vault service failure scenarios:
1. Repository access errors (network, auth)
2. Parser errors (invalid files)
3. Database errors (connection, disk space)
4. Memory errors (large vaults)

Each should return appropriate HTTP status codes:
- 503 Service Unavailable for git/database connectivity
- 507 Insufficient Storage for disk space
- 500 Internal Server Error for parser failures